app/src/main/java/com/example/epamandroid/MainActivity.kt

```diff
@@ -1,12 +1,89 @@
+import android.view.MenuItem
+import android.widget.ImageView
+import kotlinx.android.synthetic.main.activity_main.activity_main_toolbar
+import kotlinx.android.synthetic.main.activity_main.activity_main_drawer_layout
+import kotlinx.android.synthetic.main.activity_main.activity_main_nav_view
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {
+
+    private val colorList = listOf(
+            "#FFAB00",
+            "#E64A19",
+            "#00E676",
+            "#00B8D4",
+            "#01579B",
+            "#651FFF",
+            "#26C6DA",
+            "#F44336",
+            "#EF6C00"
+    )
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        configureSupportActionBar()
+
+        val navigationHeaderIconButton = activity_main_nav_view
+                .inflateHeaderView(R.layout.navigation_header)
+                .findViewById<ImageView>(R.id.navigation_header_android_icon)
+        activity_main_nav_view.setNavigationItemSelectedListener(this)
+
+        if (savedInstanceState == null) {
+            supportFragmentManager.beginTransaction().replace(R.id.activity_main_fragment_container, FirstFragment())
+                    .commit()
+            activity_main_nav_view.setCheckedItem(R.id.nav_fragment_first)
+        }
+
+        navigationHeaderIconButton.setOnClickListener {
+            navigationHeaderIconButton.setColorFilter(calculateRandomListPosition())
+        }
+
+    }
+
+    override fun onNavigationItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.nav_fragment_first -> {
+                supportFragmentManager.beginTransaction().replace(R.id.activity_main_fragment_container, FirstFragment()).commit()
+            }
+            R.id.nav_fragment_second -> {
+                supportFragmentManager.beginTransaction().replace(R.id.activity_main_fragment_container, SecondFragment()).commit()
+            }
+        }
+
+        activity_main_drawer_layout.closeDrawer(GravityCompat.START)
+
+        return true
+    }
+
+
+    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
+        return when (item?.itemId) {
+            android.R.id.home -> {
+                activity_main_drawer_layout.openDrawer(GravityCompat.START)
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun configureSupportActionBar() {
+        setSupportActionBar(activity_main_toolbar)
+
+        supportActionBar.apply {
+            this!!.setDisplayHomeAsUpEnabled(true)
+            setHomeAsUpIndicator(R.drawable.ic_menu_white)
+        }
+    }
+
+    private fun calculateRandomListPosition(): Int {
+        return parseColor(colorList[(Math.random() * 100 % (colorList.size - 1)).toInt()])
```

Comment:
please create extension for this purpose

like this or better)

![image](https://user-images.githubusercontent.com/16294951/54715229-fd001f80-4b63-11e9-9bcf-9d595d3b461c.png)